src/js/components/Select/CustomSelect.js

```diff
@@ -0,0 +1,309 @@
+                ? theme.select.custom.checkbox.check.active.background
+                : theme.select.custom.checkbox.check.active.border,
+            }}
+          >
+            {selected && (
+              <FormCheckmark {...theme.select.custom.checkbox.checkmark} />
+            )}
+          </Box>
+        </CheckBoxWrapper>
+        <Text {...theme.select.options.text}>{label}</Text>
+      </Box>
+    </OptionBox>
+  );
+}
+
+function SelectedChips({
+  options,
+  value,
+  isSelected,
+  optionLabel,
+  selectOption,
+  clearAll,
+}) {
+  const theme = useContext(ThemeContext) || defaultProps.theme;
+
+  const OptionWrapper = styled(Box)`
+    ${props => props.theme.select.custom.chips.wrapper.extend};
+  `;
+
+  const getSelectedOption = () =>
+    options.reduce((acc, item, index) => {
+      if (isSelected(index)) acc.push(index);
+      return acc;
+    }, []);
+
+  return (
+    <>
+      {Array.isArray(value) && value.length > 0 && (
+        <OptionWrapper {...theme.select.custom.chips.wrapper} wrap>
+          {getSelectedOption().map(item => (
+            <Box key={item} {...theme.select.custom.chips.option}>
+              <Text {...theme.select.custom.chips.label}>
+                {optionLabel(item)}
+              </Text>
+              <Close
+                onClick={selectOption(item)}
+                {...theme.select.custom.chips.icon}
+              />
+            </Box>
+          ))}
+          <Button focusIndicator={false} onClick={() => clearAll([])} plain>
+            <Box
+              border={{
+                side: 'bottom',
+                color: theme.select.custom.chips.clear.color,
+              }}
+            >
+              <Text {...theme.select.custom.chips.clear}>CLEAR ALL</Text>
+            </Box>
+          </Button>
+        </OptionWrapper>
+      )}
+    </>
+  );
+}
+
+function ControlButtonPanel({ onCancel, onOK }) {
+  const theme = useContext(ThemeContext) || defaultProps.theme;
+
+  const ControlButtonWrapper = styled(Box)`
+    ${props => props.theme.select.custom.controls.wrapper.extend};
+  `;
+
+  return (
+    <ControlButtonWrapper {...theme.select.custom.controls.wrapper}>
+      <Button {...theme.select.custom.controls.button} onClick={onOK} primary>
+        <Text weight={600}>OK</Text>
+      </Button>
+      <Button
+        {...theme.select.custom.controls.button}
+        onClick={onCancel}
+        secondary
+      >
+        <Text weight={600}>Cancel</Text>
+      </Button>
+    </ControlButtonWrapper>
+  );
+}
+
+function SizeBottomPanel({
+  options,
+  value,
+  isSelected,
+  optionLabel,
+  selectOption,
+  clearAll,
+  ...rest
+}) {
+  return (
+    <>
+      <SelectedChips
+        options={options}
+        value={value}
+        isSelected={isSelected}
+        optionLabel={optionLabel}
+        selectOption={selectOption}
+        clearAll={clearAll}
+      />
+      <ControlButtonPanel {...rest} />
+    </>
+  );
+}
+
+function SizeSelect({
```

Comment:
This can be just multiselect. there is no size specific use case for this component